File: projects/IsPicUs/backend-node/src/models/Image.js
import mongoose from 'mongoose';

const imageSchema = new mongoose.Schema({
  owner: { type: mongoose.Schema.Types.ObjectId, ref: 'User', index: true },
  title: { type: String, required: true },
  url: { type: String, required: true },
  thumbUrl: { type: String },
  author: { type: String },
  tags: { type: [String], index: true },
  createdAt: { type: Date, default: Date.now }
});

export default mongoose.model('Image', imageSchema);

File: projects/IsPicUs/backend-spring/src/main/java/com/ispicus/image/Image.java
package com.ispicus.image;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.List;

@Document("images")
public class Image {
  @Id
  public String id;
  @Indexed
  public String ownerId;
  public String title;
  public String url;
  public String thumbUrl;
  public String author;
  public List<String> tags;
  public Instant createdAt = Instant.now();
}

